lib/enhanced_sqlite3/adapter.rb

```diff
@@ -46,7 +46,7 @@ def configure_connection
       configure_pragmas
       configure_extensions
 
-      EnhancedSQLite3::SupportsVirtualColumns.apply!
+      EnhancedSQLite3::SupportsVirtualColumns.apply! if Gem::Version.new("7.1") >= Rails.gem_version
```

Comment:
```suggestion
      EnhancedSQLite3::SupportsVirtualColumns.apply! unless defined?(supports_virtual_columns?) && supports_virtual_columns?
```
I have been wanting to make this check condition and more intelligent. What do you think about this as the check instead?

Also, we can do the same for Deferrable Constraints:
```ruby
      EnhancedSQLite3::SupportsDeferrableConstraints.apply! unless defined?(supports_deferrable_constraints?) && supports_deferrable_constraints?
```